bigml/multivote.py

```diff
@@ -384,10 +422,42 @@ def combine(self, method=DEFAULT_METHOD, with_confidence=False,
                     raise Exception("Not enough data to use the selected "
                                     "prediction method. Try creating your"
                                     " model anew.")
-        if self.is_regression():
+        if self.boosting:
+            grouped_predictions = {}
+            for prediction in self.predictions:
+                if prediction[COMBINATION_WEIGHTS[BOOSTING]] is None:
+                    prediction[COMBINATION_WEIGHTS[BOOSTING]] = 0
+                if prediction.get(BOOSTING_CLASS) is not None:
+                    objective_class = prediction.get(BOOSTING_CLASS)
+                    if grouped_predictions.get(objective_class) is None:
+                        grouped_predictions[objective_class] = []
+                    grouped_predictions[objective_class].append(prediction)
+            # sum all gradients weighted by their "weight"
+            if self.is_regression():
+                return weighted_sum(self.predictions, weight="weight")
+            else:
+                # do the same per class and use the order of categories
+                # to decide in tie breaks
+                categories = options.get("categories", [])
+                predictions = {key: { \
+                    "probability": weighted_sum(value, weight="weight"),
+                    "order": categories.index(key)} for
+                               key, value in grouped_predictions.items()}
+                predictions = softmax(predictions)
+                prediction, prediction_info = sorted( \
+                    predictions.items(), key=lambda(x): \
+                    (- x[1]["probability"], x[1]["order"]))[0]
+                confidence = prediction_info["probability"]
+                if with_confidence:
+                    return prediction, confidence
+                elif add_confidence:
+                    return {"prediction": prediction, "confidence": confidence}
+                else:
+                    return prediction
```

Comment:
for symmetry with the other branch of the `if`, and to make this big method shorter, a possibility is to put these ~20 lines in a little function... but maybe you have reasons not to...